webapp/src/main/java/com/box/l10n/mojito/pagerduty/PagerDutyClient.java

```diff
@@ -0,0 +1,119 @@
+package com.box.l10n.mojito.pagerduty;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.text.MessageFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * PagerDuty client for creating (triggering) and resolving incidents.
+ *
+ * @author mattwilshire
+ */
+public class PagerDutyClient {
+
+  static Logger logger = LoggerFactory.getLogger(PagerDutyClient.class);
+
+  static final String BASE_URL = "https://events.pagerduty.com";
+  static final String ENQUEUE_PATH = "/v2/enqueue";
+
+  public static final int MAX_RETRIES = 3;
+  public static String exceptionTemplate =
+      "PagerDuty request failed: Status Code: '{0}', Response Body: '{1}'";
+
+  private HttpClient httpClient = HttpClient.newHttpClient();
+  private final String integrationKey;
+
+  public PagerDutyClient(String integrationKey) {
+    this.integrationKey = integrationKey;
+  }
+
+  public PagerDutyClient(String integrationKey, HttpClient httpClient) {
```

Comment:
I'm wondering if the call should always provide the client. My instincts tell me you want the HTTP client injected in as opposed to it being created by the client because config might need to be applied to the client itself (proxy headers or something)

Reply:
The client was really only injected for the unit tests so it can easily be mocked. The option is there to add your own HttpClient using this constructor however if its not needed the other constructor can be used to easily trigger incidents with the default client.

Reply:
Yeah, what I'm saying is that we should only have one constructor which has 2 args. It makes more sense

Reply:
Done, although having custom HttpClients for this is not in scope. The PagerDutyIntegrationService creates these clients from the application properties on startup, which would mean we would need the configuration for the HttpClient in there too for the mapping to be done.